src/test/java/httpServer/RequestParserTest.java

```diff
@@ -0,0 +1,105 @@
+                "Content-Length:");
+        assertFalse(requestParserHelper.atEndOfHeaders(stringBuilder));
+    }
+
+    @Test
+    void correctlyTellEndOfHeadersIsReached() throws IOException {
+        stringBuilder.append("GET / HTTP/1.1" + CRLF +
+            "Content-Length: 0" + CRLF +
+            "Host: localhost:5000" + CRLF +
+            "User-Agent: JUnit" + CRLF +
+            "Accept: */*" + CRLF + CRLF);
+        assertTrue(requestParserHelper.atEndOfHeaders(stringBuilder));
+    }
+
+    @Test
+    void storeHttpRequestWithBodyInHttpRequestObjectWorks() throws IOException {
+        ClientReadable mockBufferedReaderWrapper = new MockBufferedReaderWrapper("GET / HTTP/1.1" + CRLF +
+                "Content-Length: 2" + CRLF +
+                "Host: localhost:5000" + CRLF +
+                "User-Agent: JUnit" + CRLF +
+                "Accept: */*" + CRLF + CRLF +
+                "hi");
+        HTTPRequest request = requestParserHelper.storeHttpRequest(mockBufferedReaderWrapper);
+        assertEquals("GET", request.method);
+        assertEquals("/", request.resource);
+        assertEquals("HTTP/1.1", request.httpVersionNumber);
+        assertEquals("2", request.headers.get("Content-Length"));
+        assertEquals("localhost:5000", request.headers.get("Host"));
+        assertEquals("JUnit", request.headers.get("User-Agent"));
+        assertEquals("*/*", request.headers.get("Accept"));
+        assertEquals("hi", request.body);
+        assertEquals(4, request.headers.size());
+    }
+
+    @Test
+    void storeHttpRequestInHttpRequestObjectWorks() throws IOException {
+        ClientReadable mockBufferedReaderWrapper = new MockBufferedReaderWrapper("GET / HTTP/1.1" + CRLF +
+                "Content-Length: 0" + CRLF +
+                "Host: localhost:5000" + CRLF +
+                "User-Agent: JUnit" + CRLF +
+                "Accept: */*" + CRLF + CRLF );
+        HTTPRequest request = requestParserHelper.storeHttpRequest(mockBufferedReaderWrapper);
+        assertEquals("GET", request.method);
+        assertEquals("/", request.resource);
+        assertEquals("HTTP/1.1", request.httpVersionNumber);
+        assertEquals("0", request.headers.get("Content-Length"));
+        assertEquals("localhost:5000", request.headers.get("Host"));
+        assertEquals("JUnit", request.headers.get("User-Agent"));
+        assertEquals("*/*", request.headers.get("Accept"));
+        assertEquals(null, request.body);
+        assertEquals(4, request.headers.size());
+    }
+
+    @Test
+    void storeHttpRequestThrowsErrorForBlankRequest() throws IOException {
+        ClientReadable mockBufferedReaderWrapper = new MockBufferedReaderWrapper("");
+        boolean thrown = false;
+
+        try {
+            requestParserHelper.storeHttpRequest(mockBufferedReaderWrapper);
+        } catch (ArrayIndexOutOfBoundsException e) {
+            thrown = true;
+        }
+
+        assertTrue(thrown);
```

Comment:
This is pretty clever but there's a handy `assertThrows` available in Junit: https://www.baeldung.com/junit-assert-exception